docs/pages/guides/nextjs-starter-kit.mdx

```diff
@@ -0,0 +1,545 @@
+  );
+
+  if (usersError) {
+    return <div>Error</div>;
+  }
+
+  if (!users) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <ul>
+      {users.map((user) => (
+        <li>user.name</li>
+      ))}
+    </ul>
+  );
+}
+```
+
+### How to extend the `Document` type
+
+If you'd like to add a new property to `Document`, it's simple. First, edit the
+`Document` type in `/types/documents.ts`:
+
+```ts file="types/documents.ts"
+export type Document = {
+  // Your new property
+  randomNumber: number;
+  //...
+};
+```
+
+Then modify the return value in `/lib/server/utils/buildDocuments.ts`. This is a
+function that converts a Liveblocks room into your custom document format:
+
+```ts file="lib/server/utils/buildDocuments.ts"
+// Return our custom Document format
+const document: Document = {
+  randomNumber: Math.random(),
+  // ...
+};
+```
+
+Next, run the following command to check for problems:
+
+```
+npm run typecheck
+```
+
+If no errors are returned, the document properties were successfully extended.
+
+### How to extend the `User` & `Session` type
+
+Similar to the way we extend the `Document` type, we can also extend the `User`
+and `Session` type.
+
+Adding a new property to `User`/`Session` is simple. First, edit the `User` type
+in `/types/data.ts`.
+
+```ts file="types/data.ts"
+export type User = {
+  // Your new property
+  randomNumber: number;
+  // ...
+};
+```
+
+Then make sure to return this new property in `/lib/server/database/getUser.ts`.
+
+```ts file="lib/server/database/getUser.ts"
+return { randomNumber: Math.random() /* ... */ };
+```
+
+The new property will now be available to use in your app:
+
+```ts
+// randomNumber: Math.random(),
+console.log(session.user.info.randomNumber);
+```
+
+#### Adding this to your Liveblocks app (optional)
+
+If you'd like your new property to be available inside your Liveblocks app (e.g.
+the React hooks used in whiteboard), you must modify `UserInfo` in
+`/liveblocks.config.ts`
+
+```ts file="liveblocks.config.ts"
+export type UserInfo = Pick<User, "randomNumber" /* ... */>;
+```
+
+Then modify `/pages/api/liveblocks/auth.ts`. First, we'll give an anonymous user
+a property:
+
+```ts file="pages/api/liveblocks/auth.ts"
+// Anonymous user info
+const anonymousUser: User = {
+  randomNumber: Math.random(),
+  // ...
+};
+```
+
+Next, we'll get the signed-in user's property:
+
+```ts file="pages/api/liveblocks/auth.ts"
+// Get current user info from session (defined in /pages/api/auth/[...nextauth].ts)
+// If no session found, this is a logged out/anonymous user
+const {
+  randomNumber,
+  // ...
+} = session?.user.info ?? anonymousUser;
+```
+
+And then pass this info to `authorize`:
+
+```ts file="pages/api/liveblocks/auth.ts"
+// Get Liveblocks access token
+const { data, error } = await authorize({
+  userInfo: { randomNumber /* ... */ },
+  // ...
+});
+```
+
+To make sure to check everything's hooked up correctly:
+
+```
+npm run typecheck
```

Comment:
@taylorpwhipp with the work you're doing on TypeScript migration, are we going to explain somewhere the value of doing npm run typecheck?
Is this guide one of the first times we introduce recurrent validation as a best practice?

We could provide a small description to explain why we should run that.

I'm realizing I don't even know what it really does, even though I've used it. I know what the output (highlighting errors before running app), but not how it works and what's the whole value